datafusion-examples/examples/async_udf.rs

```diff
@@ -243,9 +252,20 @@ impl AsyncScalarUDFImpl for AsyncEqual {
         Ok(DataType::Boolean)
     }
 
+    fn invoke_with_args(
+        &self,
+        _args: ScalarFunctionArgs,
+    ) -> Result<ColumnarValue> {
+        not_impl_err!("AsyncEqual can only be called from async contexts")
+    }
+}
+
+#[async_trait]
+ impl AsyncScalarUDFImpl for AsyncEqual {
+
     async fn invoke_async_with_args(
         &self,
-        args: AsyncScalarFunctionArgs,
+        args: ScalarFunctionArgs,
```

Comment:
It is interesting here that `invoke_async_with_args has a copy of the `config_options` 🤔 -- I think that is soemthing that @Omega359  has tried to get into normal scalar functions for a while